internal/engine/compiler/engine.go

```diff
@@ -557,39 +575,38 @@ func (e *engine) CompileModule(_ context.Context, module *wasm.Module, listeners
 			}
 		}
 
+		functionEndOffset := executableOffset + len(body)
+		if executableLength < functionEndOffset {
+			if executableLength == 0 {
+				executableLength = 65536
+			}
+			for executableLength < functionEndOffset {
+				executableLength *= 2
```

Comment:
This does not ensure `executableLength >= functionEndOffset`, does it?

Also, we may want to consider if `2` is the correct const always, or if we use another number if the mapping gets really big.

Reply:
It's a loop so it will increase the size until it's large enough, isn't it? Or am I missing something?

Reply:
> Also, we may want to consider if 2 is the correct const always, or if we use another number if the mapping gets really big.

An alternative option could be to align `functionEndOffset` to the next multiple of N KiB (maybe N=64KiB).

Reply:
You're right, I mistook the shorthand `for` for an `if`. 🤦